### PR TITLE
Add Documentation and Scripts for ARO Monitor Metric testing

### PR DESCRIPTION
**Pull request by UlrichSchlueter**
### Which issue this PR addresses:

https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9534801

### What this PR does / why we need it:

Following the above WorkItem this PR contains documentation and scripts that:
- explain the monitoring data flow
- and can be used to connect a local dev monitor instance to Geneva INT 

### Test plan for issue:

Manual testing of included scripts and commands in documentation

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: docs/unit-testing-for-monitoring-metrics.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+## The Monitor Architecture
+
+The ARO monitor component (the part of the aro binary you activate when you execute ./cmd/aro monitor) collects and emits the various metrics about cluster health (and its own) we want to see in Geneva. 
+
+To send data to Geneva the monitor uses an instance of a Geneva MDM container as a proxy of the Geneva API. The MDM container accepts statsd formatted data (the Azure Geneva version of statsd, that is) over a UNIX (Domain) socket. The MDM container then forwards the metric data over a https link to the Geneva API. Please note that using a Unix socket can only be accessed from then same machine. 
+
+The monitor picks the required information about which clusters should actualyl monitor from its corresponding Cosmos DB. If multiple monitor instances run in parallel  (i.e. connect to the same database instance) as is the case in production, they negotiate which instance monitors what cluster (see : [monitoring.md](./monitoring.md)). 
+
+
+![Aro Monitor Architecture](img/AROMonitor.png "Aro Monitor Architecture")
```
Comment (Makdaam) [inline docs/unit-testing-for-monitoring-metrics.md:15]: I'd move the image to line 9 between the first and second paragraph.
From the text it's not obvious that Geneva MDM container is a different component than Geneva that serves the API. Official naming doesn't help here :/

File: hack/local-monitor-testing/deploy_MDM_VM.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ssh $CLOUDUSER@$PUBLICIP "sudo sed -i 's/SELINUX=enforcing/SELINUX=permissive/g' /etc/selinux/config"
+
+
+ssh $CLOUDUSER@$PUBLICIP "sudo firewall-cmd --zone=public --add-port=12345/tcp --permanent"
+ssh $CLOUDUSER@$PUBLICIP "sudo firewall-cmd --reload"
+
+
+scp $BASE/dockerStartCommand.sh $CLOUDUSER@$PUBLICIP:
+ssh $CLOUDUSER@$PUBLICIP "chmod +x dockerStartCommand.sh"
+ssh $CLOUDUSER@$PUBLICIP "sudo ./dockerStartCommand.sh &"
```
Comment (Makdaam) [inline hack/local-monitor-testing/deploy_MDM_VM.sh:92]: Is & intended here?

File: hack/local-monitor-testing/startMDMNetwork.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+echo "User          = $USER"
+
+VMName="$USER-mdm-link"
+CLOUDUSER="cloud-user"
+
+
+PUBLICIP=$( az vm list-ip-addresses --name $VMName -g $RESOURCEGROUP | jq -r '.[0].virtualMachine.network.publicIpAddresses[0].ipAddress' )
+
+echo "Found IP $PUBLICIP, starting socat on the mdm-link vm"
+ssh $CLOUDUSER@$PUBLICIP "sudo socat -v TCP-LISTEN:12345,fork UNIX-CONNECT:/var/etw/mdm_statsd.socket" &
```
Comment (Makdaam) [inline hack/local-monitor-testing/startMDMNetwork.sh:19]: Nit: dangling forked processes in the script. Could use information what is left running in the background and how to stop it.

**Review by Makdaam — COMMENTED** [review]
LGTM in general. I'm not sure about the pattern of leaving &-forked processes in the background without a cleanup script, but in that case it could use an additional echo explaining how to clean up.

File: docs/unit-testing-for-monitoring-metrics.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+
+### Starting the monitor
+
+When starting the monitor , make sure to have your
+
+- CLUSTER_MDM_ACCOUNT
+- CLUSTER_MDM_NAMESPACE
+  
+environment variables set to Geneva account and namespace where you metrics is supposed to land in Geneva INT (https://jarvis-west-int.cloudapp.net/)
```
Comment (cadenmarchese) [inline docs/unit-testing-for-monitoring-metrics.md:50]: Can we add a section (or maybe just a sentence) on viewing / accessing / consuming the monitor after setting these up? I assume that it's the same as getting monitor metrics in prod, but I'm not sure.

**Review by cadenmarchese — COMMENTED** [review]
Thank you for adding support and documentation for this process. I have a few first impressions questions / suggestions.

File: docs/unit-testing-for-monitoring-metrics.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+![Aro Monitor Architecture](img/AROMonitor.png "Aro Monitor Architecture")
+
+
+## Unit Testing Setup
+
+There are two ways set up: 
+- Run the Geneva container locally.
+- Spawn a VM, start the Geneva container there and connect/tunnel to it.
+
+### Local container setup
```
[the remark this answers is not in the document]
Comment (petrkotas) [reply docs/unit-testing-for-monitoring-metrics.md:24]: Hi, I agree. Prerequirements, would be great at this point. I am too a little confused. I returned to this after reading the whole.

**Review by m1kola — APPROVED** [review]
Looks good to me. Trying to make CI green now.

**Pull request merged**